### PR TITLE
Add new `SUBMITTED` status to API

### PR DESCRIPTION
**Pull request by duckception**

### CODE REVIEW

File: docs/json_rpc.md [hunk 1]
```diff
@@ -287,13 +292,26 @@ Example result:
...
+        "Status": "SUBMITTED",
+        "FinalisationBlock": null
     }
 ]
 ```
 
 ### `hubble_getBatchByHash(hash)`
 
-Returns batch information and list of included commitments in batch.
+Returns batch information and list of included commitments in batch. Commitment statuses:
```
Comment (b-tarczynski) [inline docs/json_rpc.md:311]: In this endpoint commitments don't have statuses. Maybe replace with something like:
`Returns mined or finalised batch information...`

File: api/get_batches.go [hunk 2]
```diff
@@ -19,26 +20,26 @@ func (a *API) GetBatches(from, to *models.Uint256) ([]dto.Batch, error) {
...
-func (a *API) unsafeGetBatches(from, to *models.Uint256) ([]dto.Batch, error) {
+func (a *API) unsafeGetBatches(from, to *models.Uint256) ([]interface{}, error) {
 	batches, err := a.storage.GetBatchesInRange(from, to)
 	if err != nil {
-		return []dto.Batch{}, err
+		return []interface{}{}, err
 	}
 
-	batchesWithSubmission := make([]dto.Batch, 0, len(batches))
+	batchesWithSubmission := make([]interface{}, 0, len(batches))
```
Comment (b-tarczynski) [inline api/get_batches.go:29]: We can keep  `batchesWithSubmission` type as `[]dto.Batch` as we append only objects of that type.

File: api/get_commitment_test.go [hunk 3]
```diff
@@ -105,6 +110,47 @@ func (s *GetCommitmentTestSuite) SetupTest() {
...
+			TxType:       txtype.MassMigration,
+			FromStateID:  1,
+			Amount:       models.MakeUint256(50),
+			Fee:          models.MakeUint256(10),
+			Nonce:        models.MakeUint256(0),
+			CommitmentID: &s.mmCommitment.ID,
+		},
+		SpokeID: 2,
+	}
+
```
Comment (b-tarczynski) [inline api/get_commitment_test.go:153]: Maybe extract and call `setTxs()` method instead and/or use `testutils` helper functions.

File: docs/json_rpc.md [hunk 4]
```diff
@@ -472,11 +490,44 @@ Example result (`GENESIS`):
...
+    "TransactionHash": "0xa428c43175e6156c933c436455a675c1e521e3841d9d44c6482ee26f861e49c1",
+    "SubmissionBlock": null,
+    "SubmissionTime": null,
+    "Status": "SUBMITTED",
+    "FinalisationBlock": null,
+    "AccountTreeRoot": null,
+    "Commitments": [
+        {
+            "ID": {
+                "BatchID": "4",
```
Comment (b-tarczynski) [inline docs/json_rpc.md:511]: Match `CommitmentID` with `BatchID` for consistency.

**Review by b-tarczynski — CHANGES_REQUESTED** [review]

File: models/enums/batchstatus/batch_status.go [hunk 5]
```diff
@@ -9,13 +9,13 @@ import (
 type BatchStatus uint
 
 const (
-	Pending BatchStatus = iota + 1000 // Not in use + Will be replaced in the future in favor or `Submitted`
+	Submitted BatchStatus = iota + 1001
```
[the remark this answers is not in the document]
Comment (duckception) [reply models/enums/batchstatus/batch_status.go:12]: Sure, why not. It was left this way due to historical reasons.

File: api/get_batches.go (hunk 2 above)
Comment (duckception) [reply api/get_batches.go:29]: Nice catch. It was a leftover from when I wanted to introduce `dto.SubmittedBatch` type.

File: api/get_commitment_test.go (hunk 3 above)
Comment (duckception) [reply api/get_commitment_test.go:153]: Done. I would leave further refactoring of similar problem with commitments in this `SetupTest` method for `PRO-194`.

**Review by b-tarczynski — APPROVED** [review]

**Pull request merged**